x-pack/filebeat/module/zookeeper/audit/ingest/pipeline.yml

```diff
@@ -0,0 +1,82 @@
+description: Pipeline for parsing ZooKeeper audit messages
+processors:
+- set:
+    field: event.ingested
+    value: '{{_ingest.timestamp}}'
+- rename:
+    field: message
+    target_field: log.original
+    ignore_missing: true
+- grok:
+    field: log.original
+    trace_match: true
+    patterns:
+      - '%{TIMESTAMP_ISO8601:zookeeper.audit.timestamp}%{SPACE}%{LOGLEVEL:log.level}%{SPACE}%{CALLER_CLASS:log.logger}:%{SPACE}%{GREEDYDATA:message}'
+      - '%{GREEDYDATA:message}'
+    pattern_definitions:
+      CALLER_CLASS: (%{JAVACLASS}|%{NOTSPACE})
+- rename:
+    field: '@timestamp'
+    target_field: event.created
+    if: ctx?.zookeeper?.audit?.timestamp != null
+- date:
+    if: ctx?.zookeeper?.audit?.timestamp != null && ctx.event.timezone == null
+    field: zookeeper.audit.timestamp
+    target_field: '@timestamp'
+    formats:
+    - yyyy-MM-dd HH:mm:ss,SSS
+    on_failure:
+    - append:
+        field: error.message
+        value: '{{ _ingest.on_failure_message }}'
+- date:
+    if: ctx?.zookeeper?.audit?.timestamp != null && ctx.event.timezone != null
+    field: zookeeper.audit.timestamp
+    target_field: '@timestamp'
+    formats:
+    - yyyy-MM-dd HH:mm:ss,SSS
+    timezone: '{{ event.timezone }}'
+    on_failure:
+    - append:
+        field: error.message
+        value: '{{ _ingest.on_failure_message }}'
+- kv:
+    field: message
+    field_split: "\\s+"
+    value_split: "="
+    target_field: zookeeper.audit
+- remove:
+    field:
+      - message
+      - zookeeper.audit.timestamp
+    ignore_missing: true
+- rename:
+    field: zookeeper.audit.operation
+    target_field: event.action
+    ignore_missing: true
+- set:
+    field: event.outcome
+    value: "{{zookeeper.audit.result}}"
+    ignore_empty_value: true
+    if: '["success","failure"].contains(ctx.zookeeper?.audit?.result)'
+- rename:
+    field: zookeeper.audit.ip
+    target_field: client.address
+    ignore_missing: true
+- convert:
+    field: client.address
+    target_field: client.ip
+    type: ip
```

Comment:
Note to future self; this needs 7.13 since it was added recently https://github.com/elastic/elasticsearch/pull/71285